src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -262,38 +248,37 @@ public Column()
             /// <param name="kind">The string identifier of the annotation.</param>
             /// <param name="value">Value of annotation.</param>
             /// <param name="annotationType">Type of value.</param>
-            public void AddAnnotation<T>(string kind, T value, DataViewType annotationType = null)
+            public void AddAnnotation<T>(string kind, T value, DataViewType annotationType)
             {
-                if (_annotations.ContainsKey(kind))
-                    throw Contracts.Except("Column already contains an annotation of this kind.");
-                _annotations[kind] = new AnnotationInfo<T>(kind, value, annotationType);
-            }
+                Contracts.CheckValue(kind, nameof(kind));
+                Contracts.CheckValue(annotationType, nameof(annotationType));
 
-            /// <summary>
-            /// Remove annotation from the column if it exists.
-            /// </summary>
-            /// <param name="kind">The string identifier of the annotation.</param>
-            public void RemoveAnnotation(string kind)
-            {
-                if (_annotations.ContainsKey(kind))
-                    _annotations.Remove(kind);
-                throw Contracts.Except("Column does not contain an annotation of kind: " + kind);
+                if (AnnotationInfos.ContainsKey(kind))
+                    throw Contracts.Except("Column already contains an annotation of this kind.");
+                AnnotationInfos[kind] = new AnnotationInfo<T>(kind, value, annotationType);
             }
 
             /// <summary>
             /// Returns annotations kind and type associated with this column.
             /// </summary>
             /// <returns>A dictionary with the kind of the annotation as the key, and the
             /// annotation type as the associated value.</returns>
-            public IEnumerable<KeyValuePair<string, DataViewType>> GetAnnotationTypes
+            public DataViewSchema.Annotations Annotations
             {
                 get
                 {
-                    return Annotations.Select(x => new KeyValuePair<string, DataViewType>(x.Key, x.Value.AnnotationType));
+                    var builder = new DataViewSchema.Annotations.Builder();
+                    foreach (var kvp in AnnotationInfos)
+                        builder.Add(kvp.Key, kvp.Value.AnnotationType, kvp.Value.GetGetterDelegate());
+                    return builder.ToAnnotations();
                 }
             }
         }
 
+        private SchemaDefinition()
```

Comment:
>SchemaDefinition [](start = 16, length = 16)

So, why is this private? I'm thinking about how I'd like to use it. I have my class, I create a new schema definition (but empty), then I populate the mapping. Do I have any other way to create an empty one of these guys?

Reply:
It appears that not only is there no way to create an empty one, but there is no longer a way to add a new column to it, since the column constructor is now internal.

Are we sure there are no scenarios that need to do this?